src/Attachment.tsx

```diff
@@ -8,8 +7,43 @@ import { FormatState } from './Store';
 const regExpCard = /\^application\/vnd\.microsoft\.card\./i;
 
 const YOUTUBE_DOMAIN = "youtube.com";
+const YOUTUBE_WWW_DOMAIN = "www.youtube.com";
 const YOUTUBE_SHORT_DOMAIN = "youtu.be";
+const YOUTUBE_WWW_SHORT_DOMAIN = "www.youtu.be";
 const VIMEO_DOMAIN = "vimeo.com";
+const VIMEO_WWW_DOMAIN = "www.vimeo.com";
+
+const queryStringToObject = (
+    src: string
+): any => {
+    const queryObject = {};
+    const fields = src.substr(1).split('&');
+
+    for (const field of fields) {
+        const keyValue = field.split('=');
+        queryObject[decodeURIComponent(keyValue[0])] = decodeURIComponent(keyValue[1] || '');
+    }
+
+    return queryObject;
+}
+
+const buildUrl = (
+    src: string,
+    query: Object,
+): string => {
+    const queryObject = [];
+    const builtUrl = [src, '?'];
+
+    for (const key in query) {
+        if (!query.hasOwnProperty(key)) {
```

Comment:
if you use `for... of` or `Object.keys(..)` then you don't need to filter on `hasOwnProperty`